pkg/apis/core/types.go

```diff
@@ -405,6 +405,12 @@ type PersistentVolumeClaimSpec struct {
 	// Resources represents the minimum resources required
 	// +optional
 	Resources ResourceRequirements
+	// AllocatedResources represents the resources that was requested to satisfy user
+	// request. This field is not meant to be set by the user but is used for tracking
+	// volume quota.
+	//
+	// +optional
+	AllocatedResources *ResourceRequirements
```

Comment:
If it is not meant to be set by the user, would it make more sense to have it in the `status` instead of the `spec`? Sorry if we discussed this already, I can't recall.

Reply:
The discussion from KEP - https://github.com/kubernetes/enhancements/pull/1516#discussion_r427017345

Reply:
Thank you for the reminder!

*Regarding Spec vs Status:*

You pointed to `spec.containers[i].ResourcesAllocated` (https://github.com/kubernetes/enhancements/pull/1342/files) as an example that is part of spec not status. But that design also introduces a `new admission controller [...] to limit access to ResourcesAllocated field such that only Kubelet can update allocated to a Pod, and prevents the user or any other component from changing this field.` Do you plan to do introduce such a admission controller for this field as well?


*Regarding Naming:*

You pointed to `spec.containers[i].ResourcesAllocated` (https://github.com/kubernetes/enhancements/pull/1342/files) as precedence for the name `AllocatedResources`. That name still seems like wrong to me. From our existing docs, `allocatable block indicates the amount of resources on a Node that is available to be consumed by normal Pods`. Whereas in this case it's not what is available, it's the max requested or in your words `AllocatedResources is maximum user requested size for this volume and does not reflect actual volume size of the PV`.

That said, I don't have any better suggestion, maybe API reviewers (is that @thockin @liggitt ?) can offer something better.

Reply:
> Do you plan to do introduce such a admission controller for this field as well?

This PR does not proposes such an admission controller because api-server's storage strategy is already rejecting any user applied changes.

Reply:
> This PR does not proposes such an admission controller because api-server's storage strategy is already rejecting any user applied changes.

Just to make sure I understand, how does api-server differentiate between user initiated vs controller initiated changes?

Reply:
it does not need to - basically what we are doing is, apiserver itself is only component which can set `allocatedResources` field - https://github.com/kubernetes/kubernetes/pull/91976/files#diff-46b95a53e68f2cbd22a2e756895be96eR82 . there are no controllers that are modifying this field. So when a PVC is created or updated, api-server determines if newest requested size is higher than current `allocatedResources` and updates PVC object before saving it.

Reply:
Got it, thanks for clarifying that. For the naming question, I'll leave it up to the API reviewers.